### ISSUE TITLE
Table Enhancement: Frozen column and row headers

### ISSUE THREAD

**Issue by MontyCarter** [dbeyer/benchexec#22]
It would be really nice to have tables generated with frozen or fixed column and row headers.  With this, the benchmark file row headers on the left would always be visible, in addition to the tool/execution info column headers always being visible on top.

This is by no means a feature I'm waiting on, so for my purposes it is very low priority, and simply an enhancement.  I just thought it would be nice, as we've started to generate real results, and I've noticed it's difficult to figure out which benchmark input file is causing an erroneous result when there are several runs displayed in the same table.

I haven't really looked at what's involved, and whether this is an extensive overhaul of table-generator.  If it's a big job, don't worry about it...

**Comment by PhilippWendler** [comment]
Yes, this makes a lot of sense, and it would be possible without changing the table-generator at all, just the HTML template (`template.html`). In fact, I have tried to implement this already.

The problem is that something like this is not possible with plain HTML and CSS:
Either you have a table (i.e., something in which the cells in the header align to the cells in the body),
or you have a non-scrolling header. We cannot simply set the table header to non-scrolling,
we need to use a separate tag outside the table for the header and then adjust the width of each cell in the header to the width of the corresponding column in the body using JavaScript. This can get a little bit tricky as it needs to be updated on every size change. So if somebody with experience in JavaScript would be willing to work on this, it would be great, but I cannot make any promises whether I will be able to implement this myself at some point.

**Comment by cheshire** [comment]
http://www.imaputz.com/cssStuff/bigFourVersion.html This is a non-scrolling table header, no?
The disadvantage is having to set a height in advance, but we could e.g. get user screen size with javascript and set the table height to this number.

I think we would need to separate the results into two tables though: one top "header" one (tool/limits/host/options) and the actual "results" one.
Making results one with a non-scrolling header should not be hard, I think.

**Comment by cheshire** [comment]
Actually, there is a plugin which does it out of the box: http://mkoryak.github.io/floatThead/

Opening a benchexec results page in Chrome, then typing in the javascript console: 
`$.getScript('https://cdnjs.cloudflare.com/ajax/libs/floatthead/1.2.13/jquery.floatThead.js');`

followed by

`$("#dataTable").floatThead()`

Does the trick! (everything seems to be working).
We might want to use that, but it might make sense to separate the "header" (with analysis options etc.) into the separate table, so it wouldn't scroll as well. It is also trivial to add the button to turn non-scrolling thead on/off.
Should I implement that?

**Comment by PhilippWendler** [comment]
The first link seems to assume a fixed width of every column in the table, which I think we do not want.

The second solution seems nice and we could certainly include such a script. The trick you mentioned does not work in my Firefox though (table parts start to overlap), so we need to fix that before.

I don't know however why you would want to split the table in two?

**Comment by cheshire** [comment]
>>> table parts start to overlap

Could you be more specific? you certainly want headers overlapping over the data, we just need to make them non-transparent.

>>> why split

Because the header can occupy a disproportionally large portion of the screen otherwise.

**Comment by PhilippWendler** [comment]
>>> overlap

Exactly, they were transparent in my case so this needs to be fixed.

>>> split

I see, but could you please elaborate a little bit more on your proposed solution? How would it look like? Could you create a mockup? And if we split them, aren't we back to the original problem that the column widths won't align?

**Comment by cheshire** [comment]
> Exactly, they were transparent in my case so this needs to be fixed

You mean `$("thead").css({"background": "white"})`? :P

> split

I see, maybe for know just add a button in the top navbar for toggling the fixed header on/off? that would be a fix for people with small monitors.

**Comment by cheshire** [comment]
Actually we probably want `$("#dataTable").floatThead({scrollingTop: 30});` to keep the nav-bar visible. Also the z-index / transparency for nav-bar should be fixed.

**Comment by cheshire** [comment]
OK I have added the pull request. Everything seems to work, and there is a button to toggle it on/off.

**Comment by cheshire** [comment]
OK what about now?

**Comment by cheshire** [comment]
I believe this can be closed now.
